core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,348 @@
+            change_cmd, score_category):
+        """Initializes an object of type SuggestionEditStateContent
+        corresponding to the SUGGESTION_TYPE_EDIT_STATE_CONTENT choice.
+        """
+        self.suggestion_id = suggestion_id
+        self.suggestion_type = (
+            suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT)
+        self.target_type = suggestion_models.TARGET_TYPE_EXPLORATION
+        self.target_id = target_id
+        self.target_version_at_submission = target_version_at_submission
+        self.status = status
+        self.author_id = author_id
+        self.assigned_reviewer_id = assigned_reviewer_id
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = exp_domain.ExplorationChange(change_cmd)
+        self.score_category = score_category
+
+    def validate(self):
+        """Validates a suggestion object of type SuggestionEditStateContent.
+
+        Raises:
+            ValidationError: One or more attributes of the
+                SuggestionEditStateContent object are invalid.
+        """
+        super(SuggestionEditStateContent, self).validate()
+
+        if not isinstance(self.change_cmd, exp_domain.ExplorationChange):
+            raise utils.ValidationError(
+                'Expected change_cmd to be an ExplorationChange, received %s'
+                % type(self.change_cmd))
+
+        if self.get_score_type() != suggestion_models.SCORE_TYPE_CONTENT:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be %s '
+                ', received %s' % (
+                    suggestion_models.SCORE_TYPE_CONTENT,
+                    self.get_score_type()))
+
+        if self.get_score_sub_type() not in constants.ALL_CATEGORIES:
+            raise utils.ValidationError(
+                'Expected the second part of score_category to be a valid'
+                ' category, received %s' % self.get_score_sub_type())
+
+        if self.change_cmd.cmd != exp_domain.CMD_EDIT_STATE_PROPERTY:
+            raise utils.ValidationError(
+                'Expected cmd to be %s, received %s' % (
+                    exp_domain.CMD_EDIT_STATE_PROPERTY, self.change_cmd.cmd))
+
+        if (self.change_cmd.property_name !=
+                exp_domain.STATE_PROPERTY_CONTENT):
+            raise utils.ValidationError(
+                'Expected property_name to be %s, received %s' % (
+                    exp_domain.STATE_PROPERTY_CONTENT,
+                    self.change_cmd.property_name))
+
+    def pre_accept_validate(self):
+        """Performs referential validation. This function needs to be called
+        before accepting the suggestion.
+        """
+        self.validate()
+        states = exp_services.get_exploration_by_id(self.target_id).states
+        if self.change_cmd.state_name not in states:
+            raise utils.ValidationError(
+                'Expected %s to be a valid state name' %
```

Comment:
What if I re-name the state but do not change the content (which I have actually done in the past)? This would mean a suggestion (which is still valid) would be invalidated?

Reply:
Yes the suggestion would be invalidated as it is not referentially valid anymore. 

The possible user flow that will happen here is the reviewer rejects this suggestion and then the author creates a new suggestion to the (new, renamed) state and then the reviewer can accept it.

However there is one complication of what message to show the author (as state names are only known to the creators/editors).